geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/WrapperIndex.scala

```diff
@@ -0,0 +1,47 @@
+/***********************************************************************
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.utils.index
+
+import org.locationtech.jts.geom.{Envelope, Geometry}
+import org.locationtech.jts.index.{SpatialIndex => JSTSpatialIndex}
+
+import scala.collection.JavaConverters._
+
+class WrapperIndex[T, Index <: JSTSpatialIndex](val indexBuider : () => Index) extends SpatialIndex[T] with Serializable {
+
+  var index = indexBuider()
```

Comment:
this should probably be private so we don't expose mutable state

Reply:
Changed visibility to protected: must be accessible to subclasses